source/Halibut.TestProxy/TcpTunnel.cs

```diff
@@ -50,11 +50,21 @@ public async Task Tunnel(CancellationToken cancellationToken)
                 await Task.WhenAll(
                     fromWriter.CompleteAsync().AsTask(),
                     toWriter.CompleteAsync().AsTask());
-                
+
                 Try.CatchingError(() => fromStream.Close(), e => { _logger.LogWarning("Error closing fromStream"); });
                 Try.CatchingError(() => toStream.Close(), e => { _logger.LogWarning("Error closing toStream"); });
-                Try.CatchingError(() => fromClient.Close(), e => { _logger.LogWarning("Error closing fromClient"); });
-                Try.CatchingError(() => toClient.Close(), e => { _logger.LogWarning("Error closing toClient");});
+                Try.CatchingError(() =>
+                {
+                    // Close the socket, with no waiting
+                    fromClient.Client.Close(0);
+                    fromClient.Close();
+                }, e => { _logger.LogWarning("Error closing fromClient"); });
+                Try.CatchingError(() =>
+                {
+                    // Close the socket, with no waiting
+                    toClient.Client.Close(0);
```

Comment:
can `toClient.Client.Close(0)` throw? if so is it an issue if `toClient.Close();` is not called?